xarray/core/dataarray.py

```diff
@@ -3766,6 +3769,160 @@ def bfill(self, dim: Hashable, limit: int | None = None) -> Self:
 
         return bfill(self, dim, limit=limit)
 
+    def fill_gaps(
```

Comment:
Any thoughts from others on the naming? Would `.fill` be insufficiently specific that it's filling `na`? Would `fill_missing` be clearer than `fill_gaps`?

Reply:
Open to any of those. `.fill` sounds very concise, but maybe this is easily confused with `.ffill`

Reply:
I think `.fill` could be quite nice, do others have a view?

Reply:
Maybe `gap_filler` instead, since this method does not actually fill the gaps.

I'm also wondering if its better to have a method that constructs the appropriate mask that can be used later
```
mask = ds.gap_mask(max_gap=...)
ds.ffill(...).where(~mask)
```

Reply:
`gap_filler` emphasizes the returned object type nicely! However, I choose `fill_gaps` because it fits the naming scheme of other object-returning functions better (e.g. `rolling` and `coarsen` are not called `roller` and `coarser` in xarray, even though the operation is not perfomed immediately and an object is returned).
Since I am a non-native speaker, I will follow your recommendations regarding nomenclature.

The function API is also presented as an alternative in the [initial proposal](https://github.com/pydata/xarray/issues/7665#issuecomment-1995178933). I decided to go for the object way because it is shorter (one line) and less error prone (you might easily forget the `~`). If the mask is required, you can easily get it from the object:
```python
mask=ds.gap_filler(...).mask
```